backend/mobilityAI/Controllers/SensorDataController.cs

```diff
@@ -160,9 +170,31 @@ public IActionResult AddData(IFormFile AccelerometerFile, IFormFile GyroscopeFil
             _context.GyroscopeData.AddRange(GyroscopeObjects);
             _context.SaveChanges();
 
+            
+            var accelMs = new MemoryStream();
+            AccelerometerFile.OpenReadStream().CopyTo(accelMs);
+
+            var gyroMs = new MemoryStream();
+            GyroscopeFile.OpenReadStream().CopyTo(gyroMs);
+
+            var callbackId = Guid.NewGuid().ToString();
+
+            MultipartFormDataContent form = new MultipartFormDataContent();
+
+            form.Add(new StringContent(SERVER_URL + "api/SensorData/MlCallback?Id=" + callbackId), "callback_url");
+            //form.Add(new StringContent(callbackId), "callback_id");
+            form.Add(new ByteArrayContent(accelMs.ToArray()), "file[]", AccelerometerFile.FileName);
+            form.Add(new ByteArrayContent(gyroMs.ToArray()), "file[]", GyroscopeFile.FileName);
+
+            HttpResponseMessage response = await client.PostAsync(ML_SERVER_URL + "windowify", form);
+
+            response.EnsureSuccessStatusCode();
+
+            mlCallbackIds.TryAdd(callbackId, DeviceId);
```

Comment:
What happens if this returns false (ie: a duplicate is found)?

Reply:
`TryAdd` only returns false if a [key already exists](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.tryadd?view=netframework-4.7.2) but the key is a [`Guid`](https://docs.microsoft.com/en-us/dotnet/api/system.guid?view=netcore-2.2) which is unique.